### PR TITLE
feat(monitors): Expose additional configuration

### PR DESCRIPTION
**Pull request by dcramer**
- interval schedule type
- checkin_margin
- max_runtime

### CODE REVIEW

**Comment by dcramer** [comment]
https://www.dropbox.com/s/17hefcjcj9ye8gy/Screen-Shot-2019-02-20-11-46-17.46.png?dl=1

**Comment by dcramer** [comment]
Added max_runtime to the original PR since that was low hanging on the TODO.

**Comment by dcramer** [comment]
7hm8adrw

File: src/sentry/static/sentry/app/views/organizationMonitors/monitorForm.jsx [hunk 1]
```diff
@@ -16,6 +17,19 @@ import withOrganization from 'app/utils/withOrganization';
...
 
+const SCHEDULE_TYPES = [['crontab', 'Crontab'], ['interval', 'Interval']];
+
+const MONITOR_TYPES = [['cron_job', 'Cron Job']];
+
+const INTERVALS = [
+  ['minute', 'minute(s)'],
+  ['hour', 'hour(s)'],
+  ['day', 'day(s)'],
+  ['week', 'week(s)'],
```
Comment (dcramer) [inline src/sentry/static/sentry/app/views/organizationMonitors/monitorForm.jsx]: maybe we should rethink these and just only expose minutes from a backend POV - it'd certainly make schedule configuration easier

File: src/sentry/static/sentry/app/views/organizationMonitors/monitorForm.jsx [hunk 2]
```diff
@@ -111,43 +136,99 @@ class MonitorForm extends Component {
...
-                              link: <a href="https://en.wikipedia.org/wiki/Cron" />,
-                            }
-                          )}
-                        />
-                      )
-                    );
+                    switch (this.form.getValue('config.schedule_type')) {
+                      case 'crontab':
+                        return (
+                          <div>
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/organizationMonitors/monitorForm.jsx]: Pedantic but these can probably be `<React.Fragment>`s right?

Comment (dcramer) [reply src/sentry/static/sentry/app/views/organizationMonitors/monitorForm.jsx]: Oh right forgot that existed again

File: src/sentry/static/sentry/app/views/organizationMonitors/monitorForm.jsx [hunk 3]
```diff
@@ -32,15 +46,26 @@ class MonitorForm extends Component {
   }
 
   formDataFromConfig(type, config) {
+    const rv = {};
```
Comment (dcramer) [inline src/sentry/static/sentry/app/views/organizationMonitors/monitorForm.jsx]: lol const

**Review by evanpurkhiser — APPROVED** [review]
lgtm

**Comment by dcramer** [comment]
thanks dood

**Pull request merged**